src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWING_COEFFICIENT = 1.5;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            resize();
+        }
```

Comment:
Let's collapse it in one single method.
<img width="408" alt="image" src="https://user-images.githubusercontent.com/68386737/236389084-7bf2fd50-1699-4bc8-bd20-0af8db9097a1.png">
```suggestion
            resizeIfFull() ... or ... growIfFull();
```